Add Showwcase icon

dushmanta05 opened the pull request:
<!--
Before opening your pull request, have a quick look at our contribution guidelines:
https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md

Consider adding a preview image of your submission using:
https://petershaggynoble.github.io/SI-Sandbox/preview/
-->

**Issue:** closes #6950

**Similarweb rank:** 278,711
  <!-- The Similarweb rank can be retrieved at https://www.similarweb.com
       Please see our contributing guidelines for more details on how we
       assess a brand's popularity. -->

### Checklist
  - [x] I updated the JSON data in `_data/simple-icons.json`
  - [x] I optimized the icon with SVGO or SVGOMG
  - [x] The SVG `viewbox` is `0 0 24 24`

### Description
<!--
Anything relevant, for example:
  - Why did you pick the hex value?
  - Did you manually vectorize the logo?
  - Have you used multiple sources?
  - etc.
-->

dushmanta05 commented:
Can anyone help me with the errors?

dushmanta05 commented:
There were some error in .svg file, I've updated it.

jwr12135 commented:
Hello @dushmanta05, the linter says the svg is not exactly 24px in one dimension (right now, it's 23.998 x 24.002). You should resize it in your editor to have a height (or whatever axis is larger) of exactly 24 while maintaining the aspect ratio (width should be around 23.996). Also, make sure you run it through [SVGO](https://jakearchibald.github.io/svgomg/) so it's optimized and the linter won't complain about extraneous paths.

dushmanta05 commented:
@jwr12135 Hi, thank you again for helping me out. The svg code shows I've got 24 pixels (0 0 24 24) in the code editor, even in the GitHub file. I will use another editor and see if it works. Thanks.

jwr12135 commented:
The viewbox is indeed set correctly, but the height of the `<path>` element on the inside is 24.002 and needs to be resized using an SVG editor.

dushmanta05 commented:
Hello @jwr12135, I added the height to 24 pixels but got an error again. I added height as an attribute, and the error says height is an extra attribute. Can you please look into it? Thanks.

jwr12135 commented:
All simple icons SVG's use the following format: 
```svg
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>NAME HERE</title><path d="PATH DATA HERE"/></svg>
```
So no extra height attributes are needed.

Without getting into too much detail, the path data of an SVG is a list of all the points that make up the shape. The height I'm referring to is the distance between the highest point and the lowest point. To ensure this height (or distance) is exactly 24px, we can use an SVG editor to resize the path. In the following, I show the steps needed to resize and center the SVG using [Inkscape](https://inkscape.org/).

https://user-images.githubusercontent.com/47087725/236339153-ad478bd7-78c4-4a57-a39c-7da843d2a631.mp4

jwr12135 commented on icons/showwcase.svg:
```diff
@@ -0,0 +1 @@
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Showwcase</title><path d="M8.555 1.93a14.515 14.515 0 0 1 1.695-.241c.13-.012.22-.036.273-.137.094-.168.258-.235.493-.254.62-.05 1.242-.055 1.867-.008.3.024.504.11.617.332.047.086.125.117.242.125.246.02.492.05.735.086.316.047.628.105.957.164a2.389 2.389 0 0 0-.246-.566C14.812.759 14.203.3 13.3.114c-.68-.14-1.367-.13-2.0 ...
```
```suggestion
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Showwcase</title><path d="M8.555 1.93a14.515 14.515 0 0 1 1.695-.241c.13-.012.22-.036.273-.137.094-.168.258-.235.493-.254.62-.05 1.242-.055 1.867-.008.3.024.504.11.617.332.047.086.125.117.242.125.246.02.492.05.735.086.316.047.628.105.957.164a2.389 2.389 0 0 0-.246-.566C14.812.759 14.203.3 13.3.114c-.68-.14-1.367-.13-2.055-.082a3.68 3.68 0 0 0-1.226.285c-.8.344-1.192.926-1.465 1.614ZM.48 2.549c.27.113.527.27.753.469.504.424.852.936 1.172 1.456.356.582.637 1.187.903 1.8.05-.035.054-.078.066-.117.133-.488.293-.968.496-1.437.36-.832.84-1.605 1.719-2.171.086-.055.078-.094.016-.16a5.344 5.344 0 0 0-.93-.785 1.61 1.61 0 0 0-.953-.266c-.793.012-1.516.223-2.2.527-.394.176-.761.38-1.042.684Zm10.43-.602-.085.012a.105.105 0 0 0-.035-.008c-1.039.063-2.07.164-3.07.414-1.07.266-1.897.784-2.471 1.562-.457.617-.707 1.3-.934 1.992-.238.72-.406 1.45-.55 2.187a65.356 65.356 0 0 0-.59 3.57 145.11 145.11 0 0 0-.34 2.617c-.152 1.237-.285 2.476-.293 3.722-.012 1.093.02 2.183.438 3.242a2.952 2.952 0 0 0 1.539 1.64c.824.39 1.726.606 2.647.766 1.366.234 2.753.313 4.15.332 1.494.02 2.99-.012 4.478-.187 1.09-.125 2.156-.32 3.167-.688.918-.332 1.59-.851 1.953-1.625.363-.766.488-1.566.527-2.375.098-1.824-.12-3.64-.35-5.452a98.963 98.963 0 0 0-.575-3.988 36.882 36.882 0 0 0-.47-2.425c-.218-.934-.48-1.856-.93-2.738-.55-1.075-1.47-1.835-2.873-2.16-1.746-.402-3.53-.5-5.336-.41Zm10.979 11.242c0 .106.004.21.035.313l.113.85c-.016.056-.008.107.023.177.028-.074.083-.113.067-.172l.336-1.332c.02-.05.043-.097.03-.152.068-.12.079-.254.122-.383.242-.765.41-1.543.567-2.324.17-.86.32-1.723.51-2.577a12.11 12.11 0 0 0 .294-3.18c-.015-.398-.04-.793-.172-1.207-.11.07-.203.12-.29.184-.385.293-.66.648-.905 1.02-.7 1.054-1.152 2.183-1.574 3.323-.024.06-.012.114 0 .172.223 1.034.38 2.074.53 3.117.107.722.208 1.445.314 2.17zM.068 3.48c-.07.227-.066.46-.066.688-.004 1.047.14 2.086.309 3.12.148.903.351 1.793.55 2.688.192.867.39 1.73.598 2.593.203.82.273 1.653.352 2.485.004.058-.008.124.043.186.01-.02.023-.03.023-.043.137-1.218.293-2.436.473-3.647.172-1.184.336-2.364.578-3.54.03-.151.07-.303 0-.463-.14-.317-.246-.64-.383-.957-.356-.836-.734-1.668-1.305-2.43-.23-.312-.468-.617-.84-.84-.082-.058-.175-.16-.273-.125-.086.032-.055.157-.07.243-.004.015.007.03.011.042Zm22.247-1.66a6.498 6.498 0 0 0-2.43-.444 1.13 1.13 0 0 0-.67.219c-.345.242-.583.547-.785.875-.043.066.004.094.054.129a3.7 3.7 0 0 1 .648.542c.676.714 1.02 1.546 1.305 2.398.085.262.164.524.257.816.278-.636.547-1.242.903-1.82.43-.703.89-1.39 1.7-1.87l.104-.051.09-.043c.082-.023.03-.047 0-.074-.015-.05-.058-.09-.113-.121a.42.42 0 0 0-.188-.148 2.338 2.338 0 0 0-.597-.31c-.09-.04-.172-.093-.278-.097z"/></svg>```

PeterShaggyNoble replied:
```suggestion
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Showwcase</title><path d="M8.554 2.938c.56-.103 1.126-.177 1.695-.22.128-.01.218-.031.273-.121.094-.153.26-.211.492-.228.624-.047 1.243-.05 1.869-.007.301.02.503.097.618.295.046.078.124.107.242.116.245.017.49.043.733.076.316.042.629.096.957.15a2.057 2.057 0 0 0-.244-.512c-.379-.605-.986-1.018-1.89-1.185-.679-.126-1.366-.114-2.054-.075-.43.025-.842.11-1.226.259-.8.307-1.193.831-1.466 1.452Zm-8.079.553c.272.103.527.245.757.423.503.385.849.843 1.17 1.313.356.522.637 1.067.901 1.62.053-.033.057-.07.068-.108.135-.437.293-.869.496-1.292.36-.75.841-1.444 1.718-1.953.088-.051.08-.087.015-.147a5.395 5.395 0 0 0-.927-.706 1.786 1.786 0 0 0-.955-.24c-.793.013-1.515.2-2.199.475-.394.158-.76.343-1.044.615Zm7.245-.165c-1.072.24-1.899.706-2.473 1.406-.458.557-.708 1.173-.936 1.793a15.233 15.233 0 0 0-.55 1.969 52.617 52.617 0 0 0-.589 3.215c-.122.783-.235 1.568-.34 2.354-.15 1.115-.285 2.23-.295 3.352-.01.985.021 1.966.44 2.916.274.623.746 1.135 1.54 1.477.823.353 1.724.544 2.647.688 1.366.213 2.755.283 4.148.3 1.498.016 2.994-.01 4.48-.167 1.09-.115 2.157-.29 3.168-.62.92-.3 1.59-.768 1.955-1.463.362-.69.486-1.408.528-2.136.095-1.644-.124-3.277-.354-4.91a83.408 83.408 0 0 0-.573-3.588c-.135-.73-.28-1.459-.468-2.182-.22-.84-.48-1.672-.931-2.466-.55-.967-1.473-1.65-2.876-1.944-3.4-.71-7.039-.326-8.521.006Zm14.169 9.746c0 .095.005.19.038.283l.111.766c-.016.048-.008.094.024.156.026-.064.083-.1.065-.155l.336-1.198c.023-.045.045-.089.033-.137.065-.11.08-.228.12-.342.244-.69.41-1.392.566-2.095.173-.773.321-1.55.512-2.32.236-.948.341-1.901.296-2.862-.017-.356-.04-.714-.172-1.083-.11.062-.206.108-.29.165-.386.26-.66.582-.906.915-.699.95-1.153 1.967-1.577 2.993-.021.052-.009.102.003.154.22.93.38 1.867.53 2.804.105.652.207 1.304.31 1.956zM.067 4.334c-.07.204-.066.412-.067.62-.002.941.139 1.874.308 2.807.147.81.352 1.613.551 2.417.191.78.391 1.558.6 2.335.2.739.272 1.487.348 2.236.005.054-.007.113.045.168.01-.016.023-.027.025-.039.136-1.097.292-2.191.469-3.282.172-1.064.337-2.129.58-3.186.032-.138.07-.272 0-.417-.141-.284-.247-.576-.382-.861-.357-.754-.737-1.502-1.306-2.189-.23-.278-.47-.555-.84-.754-.083-.052-.174-.145-.274-.112-.084.027-.053.141-.07.217-.002.013.008.027.013.04zm22.248-1.497a7.182 7.182 0 0 0-2.429-.4 1.25 1.25 0 0 0-.673.197c-.343.22-.58.494-.784.787-.042.06.003.084.056.116.233.139.45.303.646.49.676.643 1.022 1.392 1.307 2.16.087.234.164.47.256.734.28-.574.55-1.12.903-1.64.43-.632.893-1.249 1.699-1.682.063-.033.13-.056.195-.085.084-.022.033-.044 0-.066-.016-.048-.06-.082-.111-.112a.425.425 0 0 0-.19-.131 2.437 2.437 0 0 0-.596-.28c-.09-.035-.171-.083-.28-.088z"/></svg>
```

jwr12135 reviewed (commented):
It looks like you're pretty much good :tada:. If you accept this suggestion, it should remove the extraneous newline at the end and allow tests to pass.

dushmanta05 commented:
> It looks like you're pretty much good 🎉. You just need to remove the extraneous newline at the end, which should allow tests to pass.

Thank you for your help, that video saved the day, I'm just trying to remove the extraneous line through SVGO's tool now. Thank you again :)

dushmanta05 commented:
@jwr12135 Well, I've used SVGO tool and looking at this code, I don't think there are any newline characters but it's still failing! 
![Screenshot](https://user-images.githubusercontent.com/73304098/236502596-d8fd7656-0c3b-4f7b-838e-c034a3bc9454.png)

jwr12135 commented:
It was overly complicated, but if you merge my pull request, it should remove the new line character.

dushmanta05 commented:
Thank you so much for your help @jwr12135, this whole adding of this icon credit goes to you.

jwr12135 commented:
I don't know why the GitHub editor wouldn't let me remove the newline at the end, I had to use an editor on my computer. Hopefully for future icons you add, you'll be able to figure this out :slightly_smiling_face:.

dushmanta05 commented:
Yes, I'm going to add another new icon (that Peerlist one I was talking about in discussion). I've learned a lot from you in this pull request, and I would also like to know what editors you used. Is it VS Code?

jwr12135 commented:
Yeah, I was using vscode.

dushmanta05 commented:
Hi @jwr12135 is there anything additional I need to add for merge? Thanks

jwr12135 commented:
Actually, now that I'm looking at it again, there are differences between the original icon (in black) and yours (in red). When you were resizing the icon, did you make sure to lock the aspect ratio, before starting anything else? Just now, I grabbed the original and resized it to fit 24x24, and I got dimensions of 24x21.599; as reference, yours is 23.994x24, so the height is really stretched out.

![Screenshot_20230525_081343](https://github.com/simple-icons/simple-icons/assets/47087725/da77547e-3337-4ffe-b091-d6b4c20a7597)

dushmanta05 commented:
Thanks, @jwr12135, for responding, although my issue is closed because someone else had opened an issue with that icon earlier.

adamrusted commented on _data/simple-icons.json:
```diff
@@ -10479,6 +10479,11 @@
+        {
+            "title": "Showwcase",
+            "hex": "FFFFFF",
+            "source": "https://www.showwcase.com/"
```
```suggestion
            "source": "https://www.showwcase.com"
```

adamrusted commented on icons/showwcase.svg:
The icon [seems to have been stretched](https://github.com/simple-icons/simple-icons/assets/1830867/8d246c74-ab95-455d-86d5-e1ff7c4d35f5) to fit the 24*24 viewport. Should appear as the red version does.

PeterShaggyNoble commented:
@dushmanta05, are you still working on this one or would you like some assistance?

dushmanta05 commented:
@PeterShaggyNoble Hi! Thanks for the reminder. After the issue was closed, I thought someone else was doing it and didn't check. I'm working on it now. It would also be great if you could give me any assistance since it's my first contribution. Thanks.

dushmanta05 commented:
@PeterShaggyNoble I've updated it. Also, thank you, @adamrusted, for the suggestion :)

adamrusted reviewed (changes requested):
Thanks for taking another look at this @dushmanta05!
It looks like the icon is still stretched to _fill_ the `24*24` viewport, as opposed to only being 24px in it's widest dimension. According to Affinity Designer, it should be `24*21.6` - but centered on the `24*24` canvas.

dushmanta05 commented:
Thanks for the review, and I'm working on the logo stretched part, @adamrusted. I'll update you with further changes.

dushmanta05 commented:
@PeterShaggyNoble Thanks for the suggestion, I will check this and let you know :)

PeterShaggyNoble commented on _data/simple-icons.json:
```diff
@@ -12647,6 +12647,11 @@
         },
+        {
+            "title": "Showwcase",
+            "hex": "FFFFFF",
```
```suggestion
            "hex": "0A0D14",
```

dushmanta05 commented:
@PeterShaggyNoble Thank you so much for your effort and time, glad that you fixed the icon.
